metals/src/main/scala/scala/meta/internal/metals/PostCompileCache.scala

```diff
@@ -0,0 +1,60 @@
+package scala.meta.internal.metals
+
+import java.util.Collections.singletonList
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicBoolean
+import ch.epfl.scala.{bsp4j => b}
+import scala.meta.internal.metals.MetalsEnrichments._
+
+final class PostCompileCache(buildServer: () => Option[BuildServerConnection]) {
+  private val initialized = new AtomicBoolean(false)
+  val mainClasses = new ConcurrentHashMap[String, b.ScalaMainClass]()
+
+  def isInitialized: Boolean = initialized.get()
+
+  def initialize(target: b.BuildTargetIdentifier): CompletableFuture[Void] =
+    initialize(singletonList(target))
+
+  def initialize(
+      targets: java.util.List[b.BuildTargetIdentifier]
+  ): CompletableFuture[Void] = {
```

Comment:
We normally use `scala.concurrent.Future[Unit]` for our internal APIs and convert `.asJava` at the end before returning to lsp4j